Frameworks/Foundation/_NSInvocation.arm.asm

```diff
@@ -0,0 +1,82 @@
+;******************************************************************************
+;
+; Copyright (c) Microsoft. All rights reserved.
+;
+; This code is licensed under the MIT License (MIT).
+;
+; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+; THE SOFTWARE.
+;
+;******************************************************************************
+
+    AREA |.text|,ALIGN=4,THUMB,CODE,READONLY
+; r0 = arena start
+; r1 = frame
+; r2 = fn
+; r3 = vfp count
+    ALIGN
+|_CallFrameInternal_VFP| PROC
```

Comment:
Normally, yes, you use those macros: NESTED_ENTRY, NESTED_END, etc., and they take care of emitting the pdata and xdata into the image. With the stack pointer shenanigans going on here, though, it might not be so simple. Without investigating more deeply, I'm not sure exactly what has to be done.

---
In reply to: [76460282](https://github.com/Microsoft/WinObjC/pull/903#discussion_r76460282) [](ancestors = 76460282,76459820,76459746)